miner/Miner.hs

```diff
@@ -249,25 +255,26 @@ getWork = do
     f :: Env -> IO (Either ClientError WorkBytes)
     f e = runClientM (workClient v (chainid a) $ miner a) (ClientEnv m u Nothing)
       where
-        a = args e
+        a = envArgs e
         v = version a
-        m = mgr e
+        m = envMgr e
         u = coordinator a
 
 -- | A supervisor thread that listens for new work and manages mining threads.
 --
 mining :: (TargetBytes -> HeaderBytes -> RIO Env HeaderBytes) -> WorkBytes -> RIO Env ()
 mining go wb = do
-    race updateSignal (go tbytes hbytes) >>= traverse_ miningSuccess
+    !now <- liftIO getPOSIXTime
```

Comment:
Since the code uses the timestamp only to compute a time difference and doesn't care about absolute time, using the monotonic clock form `clock` would be more efficient.